GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -358,15 +363,14 @@ public IEnumerable<string> GetAllModifiedPaths()
             return this.modifiedPaths.GetAllModifiedPaths();
         }
 
-        public virtual void OnIndexFileChange()
+        public virtual void CheckLockWhenIndexChanged()

```

Comment:
The only purpose of this method now is to detect when the index is changed/renamed without the lock so the projection can be updated.

Reply:
I would really like the caller to be able to know which `OnIndexChanged` to call since there would be 2 if the name is kept the same.  Maybe they should be `OnIndexChangedByGit` and `OnIndexChangeDetected` or `OnIndexChangedPostCommand`?

Reply:
I like your idea of making the names `OnIndexChange[OriginOfChangeNotification]`.